### ISSUE TITLE
Reordering Main Menu Structure In NVDA

### ISSUE THREAD

**Issue by Elshara** [nvaccess/nvda#5976]
The Main menu shows 3 sub menus preferences, tools and help within NVDA. The next 4 options should be in its own sub menu called configuration or configurations. The next option, donate, should be in the help menu. The exit dialogue, should be in its own sub menu, exit and restart, exit and restart with add ons disabled, exit NVDA. This will make it much cleaner and easier as a whole.

**Comment by jcsteh** [comment]
The configuration tasks are important (and few) enough that drilling into a menu is counter-productive. Donate is at top level to give it more prominence. It would be quite non-standard to have a sub-menu for Exit and that would also be problematic because those options need to (optionally) be popped up when the user presses NVDA+q.

**Comment by PratikP1** [comment]
Agreed with Jamie regarding the donate menu item. It’s a business decision. The exit item also makes sense. I wonder whether more thought should be given regarding the last two config items. I have always believed they should be in the preferences submenu. They are certainly important but how frequently do people actually use these items?

**Comment by bhavyashah** [comment]
Hi Jamie,
I would still request rethinking for the following options requiring a
dedicated submenu, perhaps named Configuration:
Configuration profiles...
Revert to saved configuration
Reset configuration to factory defaults
Save configuration

There are four of these options, and although quite important when
needed, generally largely unused, in my experience. For a more
systematic structure and for categorization cleanness's sake, I would
be of the opinion that four not-so-frequently-used options should be
put under a Configuration submenu.
Just sharing thoughts...
Thanks.
P.S. I wholeheartedly second that Donate and Exit are perfectly well
where they already are.

-- 
Warm Regards
Bhavya Shah
Using NVDA (Non Visual Desktop Access) free and open source screen
reader for Microsoft Windows
To download a copy of the free screen reader NVDA, please visit
http://www.nvaccess.org/
Using Google Talkback on Motorolla G second generation Lollipop 5.0.2
Reach me through the following means:
Mobile: +91 7506221750
E-mail id: bhavya.shah125@gmail.com
Skype id : bhavya.09

**Comment by Elshara** [comment]
That's what motivated me to create this ticket. The configuration
menu. I added in exit menu options and donate under help menu to give
more flexibility for thought within the main menu structure itself.
Just food for thought. Although I get it is a business decision,
almost every software package I've installed other than NVDA and this
includes paid products, has had donate listed under help. It is where
I would look for it if I were to donate. The reason why I added exit
options is primarily because no other program commonly has exit option
dialogues enabled, so it makes more sense to just click on what you
want as in an exit menu. It does the same thing as a dialogue but it
is also something you can escape out of if you didn't want to
accidentally click on. In the past, exit means it's already lost. If
by chance you didn't keep the short cut on your desk top, having to
know if the cursor itself is actually on the start menu assuming
you're on windows vista and later, just to type it in, meant you may
or may not get it going again by the search command, nvda. Plus, if
you have to restart NVDA, the most basic option is especially if a
window is not responding, to use the desk top short cut as the only
option, control alt n. OF course if that fails, assuming you configure
NVDA to log on to windows screen with admin permissions, you may be
able to hit control alt delete to get shut down access to the computer
in a pinch. This of course happens in 64 bit versions of windows
rather frequently in my experience, almost always due to NVDA hanging
on some progress update or other. Having an exit menu would make
regular use a lot less cumbersome, just because if you use the same
computer as a sighted person, they may not always know where to close
down NVDA if by chance they wished to restart the computer. I know it
may seem like a lame idea, but in the long run, it would be easy
access to a layout of what you actually wanted. I only mention this
because in the tools menu, plugins are organized in a similar way.

**Comment by josephsl** [comment]
Hi,

The following response may not be related to programming, but it is nevertheless important to think about:

User experience, convenience and familiarity are all important. But what's more important is intentions as to why NV Access has placed these options in a specific order.

Donation portal: As pointed out by people, "donate" item was placed in the main menu in order to emphasize the fact that NVDA is a donation-based project (that is, NVDA follows the principle that "there is no free lunch"; in other words, to the world, NVDA is advertised as a free product, in reality, it is a partially paid screen reader in the form of voluntary donations). If NVDA was considered a general open-source project, NV Access would have placed "donate" item somewhere else, or could have mentioned it briefly in one of the dialogs. However, we need to remember that NonVisual Desktop Access is a specialized open-source project, a project designed by people with disabilities for both the general public and special audiences:

* General public: To educate the world regarding accessibility and potential people with disabilities have.
* Special audience: for blind and visually impaired people regardless of income level. We're talking about giving a second chance for our audience of 300 plus million global citizens, a significant percentage of the global population.

Thus, I'm of the opinion that "donate" item must stay where it is now.

Regarding exit options: I'd like to give a counterexample: When you wish to quit JAWS for Windows, do you go through a menu of exit/restart options, or do you press JAWS+F4 and answer affirmative when prompted? Another counterexample is Firefox: Is there a menu of options that will allow you to restart Firefox anytime you want?

The answers to these questions will let you know what is better: convenience and following trends just because it's cool, or thinking about user experience and using what's best for general cases. As a software developer, I believe it is much better to be flexible - keeping up with trends is fine for some; the real question is how to adopt it to help vast majority of the audience we are serving. Personally, I'd rather keep it as is - Window's own shutdown dialog follows this principle (the one that appears when you press Alt+F4 from desktop).

To summarize, as a person with some experience with public speaking and analyzing messages, I view each NVDA menu item (or submenus) carry a distinctive yet unified message:

* Preferences: flexibility.
* Tools: Hidden gems.
* Help: Giving assistance via various channels and glimpsing the past, present and future.
* Config profiles: New possibilities.
* Virous config settings items: Convenience place to start troubleshooting config issues.
* Donate: There is no such thing as a completely free screen reader.
* Exit: A way to close NVDA or start of a journey for troubleshooting various issues.

Because each of these items serve as a carrier of messages advertising an aspect of NVDA, moving them around could mean changing how NVDA is presented to users, developers, to the world and to our posterity. In other words, when you are asking for reordering menu structure of NVDA, what you are really asking (to me) is changing the messages NVDA carries and how NVDA is presented to the world. You could say that reordering things allows NVDA to have a fresher look, and that it will allow people to appreciate the fact that NV Access cares a lot about organization of user interface items to make it more appealing. I do agree that logical organization of interface structure is good when viewed from users' and designers' perspectives, but when we're dealing with a product that is serving as an essential companion for many people around the world (many of them who have scattered knowledge of how computers and interfaces work), I think it would not be beneficial (it could be useful now, but I see a cloudy sky down the road). Thus I would like to vote that this request be turned down for the time being - NVDA menu is a window into NVDA's power, potential and features, a carrier of various messages NVDA wishes to disclose.

Lastly, I'd like to use this ticket to address something that has been in my mind for a while: I and other NVDA contributors appreciate the fact that we have enthusiastic supporters who have provided wonderful suggestions. At the same time, I'm feeling uneasy regarding some of the suggestions we've received, some of them making us shake our heads. As I've addressed in the CAPTCHA ticket, it is better for a piece of software to become an expert in what it does, and for this expert to receive assistance from others when it is truly needed. This is one of the reasons why I appreciate the fact that we have add-ons.

The other thing I'd like to bring up is thinking critically before posting things. I do know that people have thought about these wonderful suggestions, and I'm grateful for thought-provoking conversations we had over these tickets. But I feel we could have done better - perhaps thinking about what the other side would say, possible impact of our suggestions (especially to someone living halfway around the globe who may have never heard about NVDA or just getting started with it), long-range goals, feature design and so on. This was the reason why I approached exit options like that - to let you think about it for a second, and I intend to approach other tickets like what I wrote above so we can get started on a path towards more thoughtful suggestions and bug reports. I do know that we are blunt at times, but I think what saves our face is respecting someone, especially respecting the thoughts of your audience (thus I personally become sensitive over word choice, details and so on, as I firmly believe that Github tickets are historical artifacts, a collection of public information, a public relations tool for NVDA project, and an aspect of NVDA that'll decide our reputation for years to come).

Thank you.

**Comment by Elshara** [comment]
I've been using NVDA long before the donate button was introduced. IT
should've been important then in 2006 rather than it's introduction I
believe was back in 2013 and earlier if this is to be taken seriously.
I don't particularly care how many corporate grants the software gets.
The donate page is visible in two places on the website. Free products
come with the expectancy that for normal use, all support based items
are located in one area. Otherwise, there'd be no help menu as such,
and everything would be located in one menu for easy access. What's
more, the donate button is what appears every single time someone
wants to download a copy of NVDA from the website. I think it's a
little excessive. Considering I'd be more inclined to donate if I
didn't have to see it every time I open up the NVDA menu. This is the
reason why I rarely use it, not because it exists, but because I
hardly have enough money to keep up with internet costs per ISP let
alone shell out money to projects like this. Besides, corporate funds
should keep this project going for a long while. I don't forget why
it's free, but too much focus on the fact that it's a free product
gives off the impression that other companies who offer paid software
that does the same thing are trying to shut it down. Consumers who use
it, as it is who it is targeted towards, want the ability to freely
support a project. I mean the donate button is more easy to find than
the NVDA community forums. It just sends the wrong message to people
who download it, that the pay pal button is easier to find than the
support page, which is paid support usually. I'm speaking from
experience, the best way to go about donations is to crowd source
events. I've ran websites before where I've even charged money, and it
never works out no matter how in your face it is. People start to lose
interest the second they have to see it pop up every day in their
menus. The software may be closed source, but it is hosted on source
forge, they can't even provide their own server to download it direct
from their website. And you have to skip past the donate function just
to download it. It never used to be this way from 2006 to about 2011
maybe somewhere in 2012 I mean what does it cost to host a static
website, $4.00 a month? Development costs are paid for by corporations
now as the software has grown. I just think that the way money is
poured into this project should look as professional as what it's
target audience would be inclined to use it for. Donations aside, it
makes more sense to structure things based on where people would look
to find them in a menu rather than otherwise. You don't see Skype or
Mozilla Firefox doing this and it's downloaded a lot more than NVDA. I
don't want to see it fail, quite the contrary, it's better by far than
Jaws or Window Eyes, let alone Dolphin Supernova. In fact I'm
surprised given the release of Windows 10, Microsoft isn't
incorporating it as a link in a help file on their page, but they
would Jaws. Their own narrator product does little to support any
accessibility features. So is it kick ass awesome? Yes. But little
issues like this if at least considered, would go a long way in
improving it's look as well as its use in my opinion.

**Comment by jscholes** [comment]
> Considering I'd be more inclined to donate if I didn't have to see it every time I open up the NVDA menu.

The placement of the Donate item doesn't force you to see it every time you use the menu.  If it was below Exit, i.e. the very last thing you saw, then I'd be more likely to agree.  Plus, all menu items have a convenient shortcut key, meaning that if you want to access anything, you can jump right to it without seeing any of the options.

> The software may be closed source, but it is hosted on source forge, they can't even provide their own server to download it direct
from their website.

NVDA isn't closed source, the code is right here on the GitHub repository you're using to file this issue.  The website contents and downloads haven't been hosted on SourceForge for quite some time and are now hosted by Linode.

**Comment by Elshara** [comment]
I use the short cut options frequently. I wasn't the one who said it
was closed source, the person before me said it was so I went with
that. I know you can download the source code so I figured it wasn't.
The truth is, I like source forge, but it shouldn't cost anything to
host files on a server if the server code is free to store it on such
as git hub, google code etc. I guess my point being, as I've said the
the original thread, is that I'm trying to find a way to clean up the
menu structure a bit and figured some items on it could do better in
an existing menu or as a part of a dedicated menu.

**Comment by ruifontes** [comment]
Hello!
I agree with @bhavyashah:
Donation and Exit are well as they are.

Using @josephsl description of messages send by each menu:
• Preferences: flexibility. Profiles is one of the best examples of flexibity on configurations! So, it should be in this menu...
• Tools: Hidden gems.
• Help: Giving assistance via various channels and glimpsing the past, present and future.
• Various config settings items: Convenience place to start troubleshooting config issues. Agree, but they should be in a menu as items of that menu...
• Donate: There is no such thing as a completely free screen reader.
• Exit: A way to close NVDA or start of a journey for troubleshooting various issues.

And that way, we will have only 6 items in NVDA main menu...
